### PR TITLE
[Merged by Bors] - feat: port GroupTheory.FreeAbelianGroupFinsupp

### PR DESCRIPTION
**Pull request by int-y1**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by int-y1** [comment]
todo:

* [ ] `toFinsupp_comp_toFreeAbelianGroup` error

File: Mathlib/GroupTheory/FreeAbelianGroupFinsupp.lean [hunk 1]
```diff
@@ -54,53 +54,58 @@ theorem Finsupp.toFreeAbelianGroup_comp_singleAddHom (x : X) :
...
 @[simp]
 theorem FreeAbelianGroup.toFinsupp_comp_toFreeAbelianGroup :
     toFinsupp.comp toFreeAbelianGroup = AddMonoidHom.id (X →₀ ℤ) := by
-  ext (x y); simp only [AddMonoidHom.id_comp]
+  apply addHom_ext'
+  intro x
+  apply AddMonoidHom.ext_int
+  apply ext
+  intro y
+  simp only [AddMonoidHom.id_comp]
```
Comment (eric-wieser) [inline Mathlib/GroupTheory/FreeAbelianGroupFinsupp.lean]: This looks like an underlying bug that we should fix. At the very least we should have added a porting note here

**Review by eric-wieser — COMMENTED** [review]
It looks to me like an `ext` priority is missing, which is why you've had to mangle the `ext` tactic when porting.

I think we should track down the missing lemma and fix it, rather than merging the hacks here.

File: Mathlib/GroupTheory/FreeAbelianGroupFinsupp.lean (hunk 1 above)
Comment (Parcly-Taxel) [reply Mathlib/GroupTheory/FreeAbelianGroupFinsupp.lean]: All guilty `ext` lemmas have been found out and given a high priority.

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**